hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -487,6 +497,70 @@ waitForLog delay nodeOutput failureMessage predicate = do
         ]
           <> logs
 
+timedTx :: FilePath -> Tracer IO EndToEndLog -> RunningNode -> TxId -> IO ()
+timedTx tmpDir tracer node@RunningNode{networkId, nodeSocket} hydraScriptsTxId = do
+  (aliceCardanoVk, aliceCardanoSk) <- keysFor Alice
+  let aliceSk = generateSigningKey "alice-timed"
+  let alice = deriveParty aliceSk
+  let contestationPeriod = UnsafeContestationPeriod 2
+  aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket [] contestationPeriod
+  withHydraNode tracer aliceChainConfig tmpDir 1 aliceSk [] [1] hydraScriptsTxId $ \n1 -> do
+    waitForNodesConnected tracer [n1]
+    let lovelaceBalanceValue = 100_000_000
+
+    -- Funds to be used as fuel by Hydra protocol transactions
+    seedFromFaucet_ node aliceCardanoVk lovelaceBalanceValue Fuel (contramap FromFaucet tracer)
+    send n1 $ input "Init" []
+    headId <-
+      waitForAllMatch 10 [n1] $
+        headIsInitializingWith (Set.fromList [alice])
+
+    -- Get some UTXOs to commit to a head
+    committedUTxOByAlice <- seedFromFaucet node aliceCardanoVk aliceCommittedToHead Normal (contramap FromFaucet tracer)
+    send n1 $ input "Commit" ["utxo" .= committedUTxOByAlice]
+    waitFor tracer 3 [n1] $ output "HeadIsOpen" ["utxo" .= committedUTxOByAlice, "headId" .= headId]
+
+    -- Acquire a current point in time
+    genesisParams <- queryGenesisParameters networkId nodeSocket QueryTip
+    let slotLengthSec = protocolParamSlotLength genesisParams
+    currentSlot <- queryTipSlotNo networkId nodeSocket
+
+    -- Create an arbitrary transaction using some input.
+    let firstCommittedUTxO = Prelude.head $ UTxO.pairs committedUTxOByAlice
+
+    -- Create a transaction which is only valid in 5 seconds
+    let
+      secondsToAwait = 15
+      slotsToAwait = SlotNo . truncate $ fromInteger secondsToAwait / slotLengthSec
+      futureSlot = currentSlot + slotsToAwait
+      lovelaceToSend = lovelaceBalanceValue - 90_000_000
+
+      -- TODO (later) use time in a script (as it is using POSIXTime)
+      Right tx =
+        mkRangedTx
+          firstCommittedUTxO
+          (inHeadAddress aliceCardanoVk, lovelaceToValue lovelaceToSend)
+          aliceCardanoSk
+          (Just $ TxValidityLowerBound futureSlot, Nothing)
+
+    -- First submission: invalid
+    send n1 $ input "NewTx" ["transaction" .= tx]
+    waitMatch 3 n1 $ \v -> do
+      guard $ v ^? key "tag" == Just "TxInvalid"
+
+    -- Wait for the future chain slot and time
+    threadDelay 15
```

Comment:
Should: use the variable from above to wait for the right amount of time

Also.. 15 seconds is a bit long for a happy path execution